arch/xtensa/src/esp32s2/esp32s2_spi.c

```diff
@@ -0,0 +1,1231 @@
+  esp32s2_gpiowrite(config->mosi_pin, true);
+  esp32s2_gpiowrite(config->miso_pin, true);
+  esp32s2_gpiowrite(config->clk_pin, true);
+
+#if SPI_HAVE_SWCS
+  esp32s2_configgpio(config->cs_pin, OUTPUT_FUNCTION_1);
+  esp32s2_gpio_matrix_out(config->cs_pin, SIG_GPIO_OUT_IDX, 0, 0);
+#endif
+
+  /* SPI3 doesn't have IOMUX, if SPI3 is enabled use GPIO Matrix for both */
+
+  if (esp32s2_spi_iomux(priv))
+    {
+#if !SPI_HAVE_SWCS
+      esp32s2_configgpio(config->cs_pin, OUTPUT_FUNCTION_5);
+      esp32s2_gpio_matrix_out(config->cs_pin, SIG_GPIO_OUT_IDX, 0, 0);
+#endif
+      esp32s2_configgpio(config->mosi_pin, OUTPUT_FUNCTION_5);
+      esp32s2_gpio_matrix_out(config->mosi_pin, SIG_GPIO_OUT_IDX, 0, 0);
+
+      esp32s2_configgpio(config->miso_pin, INPUT_FUNCTION_5 | PULLUP);
+      esp32s2_gpio_matrix_out(config->miso_pin, SIG_GPIO_OUT_IDX, 0, 0);
+
+      esp32s2_configgpio(config->clk_pin, OUTPUT_FUNCTION_5);
+      esp32s2_gpio_matrix_out(config->clk_pin, SIG_GPIO_OUT_IDX, 0, 0);
+    }
+  else
+    {
+#if !SPI_HAVE_SWCS
+      esp32s2_configgpio(config->cs_pin, OUTPUT);
+      esp32s2_gpio_matrix_out(config->cs_pin, config->cs_outsig, 0, 0);
+#endif
+      esp32s2_configgpio(config->mosi_pin, OUTPUT);
+      esp32s2_gpio_matrix_out(config->mosi_pin, config->mosi_outsig, 0, 0);
+
+      esp32s2_configgpio(config->miso_pin, INPUT | PULLUP);
+      esp32s2_gpio_matrix_in(config->miso_pin, config->miso_insig, 0);
+
+      esp32s2_configgpio(config->clk_pin, OUTPUT);
+      esp32s2_gpio_matrix_out(config->clk_pin, config->clk_outsig, 0, 0);
+    }
+
+  modifyreg32(SYSTEM_PERIP_CLK_EN0_REG, 0, config->clk_bit);
+  modifyreg32(SYSTEM_PERIP_RST_EN0_REG, config->rst_bit, 0);
+
+  regval = SPI_DOUTDIN_M | SPI_USR_MISO_M | SPI_USR_MOSI_M | SPI_CS_HOLD_M;
+  putreg32(regval, SPI_USER_REG(priv->config->id));
+  putreg32(0, SPI_USER1_REG(priv->config->id));
+  putreg32(0, SPI_SLAVE_REG(priv->config->id));
+  putreg32(SPI_CS1_DIS_M | SPI_CS2_DIS_M,
+           SPI_MISC_REG(priv->config->id));
+
+#if SPI_HAVE_SWCS
+  esp32s2_spi_set_regbits(SPI_MISC_REG(priv->config->id), SPI_CS0_DIS_M);
+#endif
+
+  putreg32(0, SPI_CTRL_REG(priv->config->id));
+  putreg32(VALUE_MASK(0, SPI_CS_HOLD_TIME),
+           SPI_USER1_REG(priv->config->id));
+
+  esp32s2_spi_setfrequency(dev, config->clk_freq);
+  esp32s2_spi_setbits(dev, config->width);
+  esp32s2_spi_setmode(dev, config->mode);
+}
+
+/****************************************************************************
+ * Name: esp32s2_spi_deinit
+ *
+ * Description:
+ *   Deinitialize ESP32-S2 SPI hardware interface.
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *
+ * Returned Value:
+ *   None.
+ *
+ ****************************************************************************/
+
+static void esp32s2_spi_deinit(struct spi_dev_s *dev)
+{
+  struct esp32s2_spi_priv_s *priv = (struct esp32s2_spi_priv_s *)dev;
+
+  modifyreg32(SYSTEM_PERIP_RST_EN0_REG, 0, priv->config->clk_bit);
```

Comment:
```suggestion
  modifyreg32(SYSTEM_PERIP_RST_EN0_REG, 0, priv->config->rst_bit);
```
This is also wrong on the `esp32`, `esp32c3` and `esp32s3` chips, could you take the opportunity to fix them as well?
Luckily both `clk_bit` and `rst_bit` refer to `BIT(6)` from their respective registers, so there won't be any real impact.